clusterloader2/pkg/measurement/common/simple/wait_for_controlled_pods.go

```diff
@@ -303,29 +303,63 @@ func (w *waitForControlledPodsRunningMeasurement) handleObject(clients *framewor
 
 	if err := w.deleteObjectLocked(clients.GetClient(), oldRuntimeObj); err != nil {
 		glog.Errorf("%s: delete checker error: %v", w, err)
-	} else if newRuntimeObj == nil {
-		if err := w.handleObjectLocked(clients.GetClient(), oldRuntimeObj, true); err != nil {
-			glog.Errorf("%s: create checker error: %v", w, err)
-		}
 	}
-	if err := w.handleObjectLocked(clients.GetClient(), newRuntimeObj, false); err != nil {
+	if err := w.handleObjectLocked(clients.GetClient(), oldRuntimeObj, newRuntimeObj); err != nil {
 		glog.Errorf("%s: create checker error: %v", w, err)
 	}
 }
 
-func (w *waitForControlledPodsRunningMeasurement) handleObjectLocked(c clientset.Interface, obj runtime.Object, isDeleted bool) error {
-	if obj == nil {
+func checkScaledown(oldObj, newObj runtime.Object) (bool, error) {
+	if newObj == nil {
+		return true, nil
+	}
+	if oldObj == nil {
+		return false, nil
+	}
```

Comment:
Instead of having those if`s, can you please:
- add "if obj == nil return 0" statement to GetReplicasFromRuntimeObject function (similarly as it is handled in GetSpecFrom..
- remove those if`s here

Reply:
Done.